pkg/cmd/cli/secrets/new.go

```diff
@@ -212,19 +220,46 @@ func (o *CreateSecretOptions) BundleSecret() (*kapi.Secret, error) {
 	return secret, nil
 }
 
-func readFile(filePath string, dataMap map[string][]byte) error {
-	fileName := path.Base(filePath)
-	if !kvalidation.IsSecretKey(fileName) {
-		return fmt.Errorf("%s cannot be used as a key in a secret", filePath)
+func addKeyToSecret(keyName, filePath string, secretData map[string][]byte) error {
+	if _, entryExists := secretData[keyName]; entryExists {
+		return fmt.Errorf("Cannot add key %s from path %s, another key by that name already exists.", keyName, filePath)
+	}
+	data, err := readFile(filePath)
+	if err != nil {
+		return err
 	}
-	if _, exists := dataMap[fileName]; exists {
-		return fmt.Errorf("Multiple files with the same name (%s) cannot be included in a secret", fileName)
+	secretData[keyName] = data
+}
+
+// parseSource parses the source given. Acceptable formats include:
+// source-name=source-path, where source-name will become the key name and source-path is the path to the key file
+// source-path, where source-path is a path to a file or directory, and key names will default to file names
+// Key names cannot include '='.
+func parseSource(source string) (keyName, filePath string, err error) {
+	numSeparators := strings.Count(source, "=")
+	switch {
+	case numSeparators == 0:
+		return path.Base(source), source, nil
```

Comment:
This skips validation of the key being returned and that's the only place you're checking.